### ISSUE TITLE
Remember card option

### ISSUE THREAD

**Issue by jvorcak** [braintree/braintree-web-drop-in#470]
### General information

* SDK/Library version: 1.14.1
* Environment: Sandbox

### Related isues
https://stackoverflow.com/questions/47815910/braintree-drop-in-ui-disable-adding-new-payment-method
https://github.com/braintree/braintree-android-drop-in/issues/39

### Issue description

I think it would be a very useful thing to have an option for showing "Remember me" checkbox. 

It seems uncommon to automatically save a card without an explicit request from a user. In my use case, I want to associate a payment with my customer, so I want to provide his/her id whene generating a token.
I'm OK with storing a card on Braintree side, all I want is for my user to explicitly ask for it.

Do you think this could be considered? From what I see, other people would find it useful as well.

**Comment by jordan-brough** [comment]
**We have the same request**: Our customers would sometimes like to be able to opt out of vaulting their credit cards.

Our customers often want us to vault their credit cards to make checkout easier. But a significant number of customers sometimes do not want a particular credit card vaulted (or even are not allowed to have that particular card vaulted as per their company policy).

i.e., we are looking for a "Save this card" option, which seems fairly standard on ecommerce websites in general. (I spent a few minutes verifying this and I found multiple large ecommerce sites with exactly this sort of option).

Is this a feature that Braintree would be willing to add or to accept a PR for? Is it possible to add this functionality purely via this repo, or would it require changes to Braintree's backend?

We would like to be able to offer this option to our customers without having to roll our own solution via hosted fields.

We've talked with Braintre support a bit and they've described a few workarounds but all of them have drawbacks that we'd rather avoid.

**Comment by jordan-brough** [comment]
@jvorcak I noticed your comment here: https://github.com/braintree/braintree-android-drop-in/issues/39#issuecomment-458039179

Is that working as expected for you?

It looks like the relevant code is here: https://github.com/braintree/braintree-web-drop-in/blob/a23cd3e76764d5c2a4849b32736f46c0a11ef4c1/src/views/payment-sheet-views/card-view.js#L351-L353

If that is all it takes, then it sounds like this is possible to do purely client side, and Braintree server-side updates would not be required?

Braintree - would you be able to add this as a first-class option in this repo? Perhaps a configuration option like "showSaveCardCheckbox" (or something similar) that allows showing a "save this card" option?  And has the effect of forcing `vault: true` when tokenizing the payment method?

**Comment by jvorcak** [comment]
@jordan-brough glad to hear that you are in the same situation :)
Yeah, it seems to work for me and it does exactly what I need. However, it's quite hackish. My guess is that a client only PR could fix this, but I'm not sure - I've only spent 1-2 days playing with Braintree so far.

**Comment by jordan-brough** [comment]
Super, thanks @jvorcak!

I'd love to see this added as an explicit option in the drop-in with separate storage -- not having to stuff it into `isGuestCheckout` and not having to do it in a hacky way.

Perhaps a configuration option like:
```js
braintree.create({ … , showVaultCheckbox: true })
```
?

And when set to true a checkbox would be exposed within the drop-in UI as something like "[ ] save this card".

And then the value could be used with something like this:
(in `src/views/payment-sheet-views/card-view.js`)
```js
CardView.prototype.tokenize = function () {
  …
  var tokenizeOptions = {
    vault: !self.model.isGuestCheckout && !vaultingDisabled()
  };
  …
}

function vaultingDisabled() {
  // note: this is just pseudocode
  config.showVaultCheckbox && !saveCardCheckbox.isSelected()
}
```

Braintree any thoughts on this?

**Comment by crookedneighbor** [comment]
This is something we have planned for the next major version of Drop-in. We'll discuss internally if we can squeeze it in to the current version.

**Comment by crookedneighbor** [comment]
In progress here: https://github.com/braintree/braintree-web-drop-in/pull/481

**Comment by Sravanchittimalla** [comment]
Hi Everyone,

This is a really cool feature if they add it by default. 
Else, you can add a checkbox by your self other than drop-in UI and the on the server side while processing the transaction you can pass the value in options.

gateway.transaction.sale({
  amount: "10.00",
  paymentMethodNonce: nonceFromTheClient,
  customerId: "theCustomerId",
  options: {
    storeInVaultOnSuccess: true //If this is set to true the this will save the data on payment success.
  }
}, function (err, result) {
});

Here is the URL: https://developers.braintreepayments.com/reference/request/transaction/sale/node

**Comment by hollabaq86** [comment]
👋 we just released version [1.18.0](https://github.com/braintree/braintree-web-drop-in/blob/master/CHANGELOG.md#1180), which includes optional [card vaulting](https://braintree.github.io/braintree-web-drop-in/docs/current/module-braintree-web-drop-in.html#~cardCreateOptions)! 🎉 I'm going to go ahead and close this issue, but please let us know if you encounter any issues.

**Comment by bhushan90** [comment]
How to conditionally vault the payment method based on checkbox I didn't find any resource on that related to drop-in UI

**Comment by rossiter10** [comment]
> 👋 we just released version [1.18.0](https://github.com/braintree/braintree-web-drop-in/blob/master/CHANGELOG.md#1180), which includes optional [card vaulting](https://braintree.github.io/braintree-web-drop-in/docs/current/module-braintree-web-drop-in.html#~cardCreateOptions)! 🎉 I'm going to go ahead and close this issue, but please let us know if you encounter any issues.

So this was only added to v3 of the javascript SDK, right? It wouldn't be available in v2?

**Comment by crookedneighbor** [comment]
@rossiter10 it was only added to this version of Drop-in (which uses the underlying v3 JS SDK), it was not and will not be available in the version of Drop-in bundled into v2 of the JS SDK.
